[BUG] test_non_empty_ctas fails on yarn

**Describe the bug**
06:31:10  integration_tests/src/main/python/parquet_write_test.py::test_non_empty_ctas[True][ALLOW_NON_GPU(DataWritingCommandExec,HiveTableScanExec)] FAILED [ 99%]
06:31:10  integration_tests/src/main/python/parquet_write_test.py::test_non_empty_ctas[False][ALLOW_NON_GPU(DataWritingCommandExec,HiveTableScanExec)] FAILED [100%]
06:31:10  

```
E                   py4j.protocol.Py4JJavaError: An error occurred while calling o106.sql.
06:31:10  E                   : java.lang.ClassNotFoundException: org.apache.hadoop.hive.ql.plan.LoadTableDesc$LoadFileType
06:31:10  E                   	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
06:31:10  E                   	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
06:31:10  E                   	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
06:31:10  E                   	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
06:31:10  E                   	at org.apache.****.sql.hive.client.Shim_v3_0.clazzLoadFileType$lzycompute(HiveShim.scala:1272)
06:31:10  E                   	at org.apache.****.sql.hive.client.Shim_v3_0.clazzLoadFileType(HiveShim.scala:1271)
06:31:10  E                   	at org.apache.****.sql.hive.client.Shim_v3_0.loadTable(HiveShim.scala:1355)
06:31:10  E                   	at org.apache.****.sql.hive.client.HiveClientImpl.$anonfun$loadTable$1(HiveClientImpl.scala:885)
06:31:10  E                   	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
06:31:10  E                   	at org.apache.****.sql.hive.client.HiveClientImpl.$anonfun$withHiveState$1(HiveClientImpl.scala:294)
06:31:10  E                   	at org.apache.****.sql.hive.client.HiveClientImpl.liftedTree1$1(HiveClientImpl.scala:227)
06:31:10  E                   	at org.apache.****.sql.hive.client.HiveClientImpl.retryLocked(HiveClientImpl.scala:226)
06:31:10  E                   	at org.apache.****.sql.hive.client.HiveClientImpl.withHiveState(HiveClientImpl.scala:276)
06:31:10  E                   	at org.apache.****.sql.hive.client.HiveClientImpl.loadTable(HiveClientImpl.scala:880)
06:31:10  E                   	at org.apache.****.sql.hive.HiveExternalCatalog.$anonfun$loadTable$1(HiveExternalCatalog.scala:884)
06:31:10  E                   	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
06:31:10  E                   	at org.apache.****.sql.hive.HiveExternalCatalog.withClient(HiveExternalCatalog.scala:103)
06:31:10  E                   	at org.apache.****.sql.hive.HiveExternalCatalog.loadTable(HiveExternalCatalog.scala:878)
06:31:10  E                   	at org.apache.****.sql.catalyst.catalog.ExternalCatalogWithListener.loadTable(ExternalCatalogWithListener.scala:167)
06:31:10  E        
```

--- a/integration_tests/src/main/python/parquet_write_test.py
+++ b/integration_tests/src/main/python/parquet_write_test.py
@@ -325,6 +325,7 @@ def test_non_empty_ctas(spark_tmp_path, spark_tmp_table_factory, allow_non_empty
     data_path = spark_tmp_path + "/CTAS"
     conf = {
         "spark.sql.hive.convertCTAS": "true",
+        "spark.sql.legacy.createHiveTableByDefault": "false",
         "spark.sql.legacy.allowNonEmptyLocationInCTAS": str(allow_non_empty)
     }
     def test_it(spark):

